src/KeyManagement.sol

```diff
@@ -16,10 +17,24 @@ abstract contract KeyManagement is IKeyManagement {
     function _onlyThis() internal view virtual {}
 
     /// @inheritdoc IKeyManagement
-    function authorize(Key memory key) external returns (bytes32 keyHash) {
+    function register(Key memory key) external {
         _onlyThis();
-        keyHash = _authorize(key);
-        emit Authorized(keyHash, key);
+
+        bytes32 keyHash = key.hash();
+        // If the keyHash already exists, it does not revert and updates the key instead.
```

Comment:
Can just early return if `keyHashes.contains(keyHash)`?

Reply:
this is more expensive in the register case where it hasn't been registered yet though